spec/classes/aerospike_spec.rb

```diff
@@ -413,6 +413,22 @@
         # We still manage the config file
         it { is_expected.to create_file('/etc/aerospike/aerospike.conf') }
       end
+
+      describe 'allow changing service provider' do
+        let(:params) {{
+          :service_provider => 'systemd',
+        }}
+        let(:facts) {{
+          :osfamily => osfamily,
+        }}
+
+        it { is_expected.to compile.with_all_deps }
+        it { is_expected.to contain_class('aerospike::install').that_comes_before('Class[aerospike::config]') }
+        it { is_expected.to contain_class('aerospike::config').that_comes_before('Class[aerospike::service]') }
+        it { is_expected.to contain_class('aerospike::service') }
+
+        it { is_expected.to contain_service('aerospike') }
```

Comment:
I don't see any test on the provider itself here. Maybe it misses a `.with_provider('systemd')` ?